subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -49,6 +49,13 @@ LOG_MODULE_REGISTER(nrf_cloud_coap_transport, CONFIG_NRF_CLOUD_COAP_LOG_LEVEL);
 #define VER_STRING_FMT "mver=%s&cver=%s&dver=%s"
 #define VER_STRING_FMT2 "cver=" CDDL_VERSION "&dver=" BUILD_VERSION_STR
 #define BUILD_VERSION_STR STRINGIFY(BUILD_VERSION)
+#define COAP_PROXY_RSC "proxy"
+
+/* CoAP extended options: When multiple options are used,
+ * for simplicity, always put COAP_OPTION_ACCEPT first
+ */
+#define ACPT_IDX 0 /* COAP_OPTION_ACCEPT */
+#define PRXY_IDX 1 /* COAP_OPTION_PROXY_URI */
```

Comment:
I fear this will lead to confusion since the underlying options don't seem to care about the order they are provided in

Reply:
As a compromise, we could assert that `opt->value[ACPT_IDX].code == COAP_OPTION_ACCEPT` in client_transfer_opt

Though I see now that it is a static function, so I guess we only stand to confuse our future selves

Reply:
the only reason i added this was for the `LOG_DBG` on line 552.

Reply:
CoAP options need to be specified in numerical order or else you'll get a message (when `CONFIG_LOG_NET=y` and CONFIG_COAP_LOG_LEVEL_DBG=y`) indicating the "Option is not added in ascending order".

It turns out this is a requirement for encoding the option number.  See:
https://datatracker.ietf.org/doc/html/rfc7252#section-3.1

The Zephyr CoAP stack does the right thing and inserts the out-of-order option in the correct position, but it is best to provide them in order.